homeassistant/components/apache_kafka/__init__.py

```diff
@@ -32,6 +36,9 @@
                 vol.Required(CONF_PORT): cv.port,
                 vol.Required(CONF_TOPIC): cv.string,
                 vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+                vol.Optional(CONF_SECURITY_PROTOCOL, default="PLAINTEXT"): cv.string,
+                vol.Optional(CONF_USERNAME, default=""): cv.string,
```

Comment:
Don't need a default if it's going to be falsy – `vol.Optional` will default to `None`.

Reply:
Nice catch!

Reply:
@bachya the parameter requires a string, if we remove the default we'll have to pass in `username` and `password` field in config yaml in order to avoid `key_error`. While having a default doesn't break existing implementation + Does the job in the simplest way IMO. LMK what you think

Thanks!

Reply:
The library defaults the username/password to `None`:

<https://github.com/aio-libs/aiokafka/blob/ea1aab8f8d6d83a6410a9761f6b4326e3b1283e4/aiokafka/producer/producer.py#L189>

The default can be removed, as suggested by @bachya.

When getting the value instead of the current `conf[CONF_USERNAME]` use `conf.get(CONF_USERNAME)`.

Reply:
@frenck My bad. Got it. Thanks!